### DISCUSSION TITLE
How to keep TQDM progress bars for different iterations on the same line?

### DISCUSSION THREAD

**Discussion by seanswyi** [tqdm/tqdm#1141] [Q&A]
I'm running some code and have TQDM used like this:

```
data = some_data()

for epoch in range(num_epochs):
    progress_bar = tqdm(iterable=data, desc=f"Training {epoch + 1}/{int(num_epoch)}", total=len(data))

    for step, batch in enumerate(progress_bar):
        .
        .
        .
```

It's the same for my evaluation code below.

Running the script produces something like this:
[![enter image description here][1]][1]

What usually happens is that the evaluation progress bars will fill up beneath the current training bar, and then move up with a new training bar starting at the very bottom.

When the entire process finishes, I have a total of `num_epochs * 3` progress bars on my screen, and I would like to only have three with each being updated in place.

How should I go about doing this?

  [1]: https://i.stack.imgur.com/SiVNW.png

**Comment by casperdcl** [comment]
sounds like you need `leave=False` (or `leave=epoch==num_epochs-1`)

**Reply by seanswyi** [reply]
Thanks for the reply. Unfortunately this doesn't achieve what I want. The bars are not being updated in-place and are still printing to standard output.

**Reply by sascharo** [reply]
> Unfortunately, this doesn't achieve what I want. The bars are not being updated in-place and are still printing to standard output.

Did you find a solution?

**Comment by hetari** [comment]
Would you like to clear the console? Although not the most efficient solution, it should work.
```py
# Your code before printing the progress bar...
import os
os.system('cls' if os.name == 'nt' else 'clear')
# printing the progress bar code goes here...
```
